test/test_objectives.py

```diff
@@ -20,20 +22,54 @@
 )
 
 
+""" Tests for distance objective """
+
+
 @pytest.mark.parametrize(
-    "method",
+    "method,max_motion_cost",
     [
-        objectives.DistanceMethod.EUCLIDEAN,
-        objectives.DistanceMethod.LATLON,
-        objectives.DistanceMethod.OMPL_PATH_LENGTH,
+        (objectives.DistanceMethod.EUCLIDEAN, 2.5),
+        (objectives.DistanceMethod.LATLON, 2700),
+        (objectives.DistanceMethod.OMPL_PATH_LENGTH, 4.0),
     ],
 )
-def test_distance_objective(method: objectives.DistanceMethod):
+def test_distance_objective(method: objectives.DistanceMethod, max_motion_cost: float):
     distance_objective = objectives.DistanceObjective(
         PATH._simple_setup.getSpaceInformation(),
         method,
     )
-    assert distance_objective is not None
+
+    # test sample_states()
+    num_samples = 3
+    sampled_states = distance_objective.sample_states(num_samples)
+    assert len(sampled_states) == num_samples
+    # for state in sampled_states:
+    #     assert ompl_path.is_state_valid(state)
+
+    # test find_maximum_motion_cost()
+    # implicitly
+    assert distance_objective.max_motion_cost == pytest.approx(max_motion_cost, rel=1e0)
+    # explicitly for the easiest method to set up
+    # don't need to test for all methods since they each have their own tests
+    if method == objectives.DistanceMethod.OMPL_PATH_LENGTH:
+        states = []
+        space = PATH._simple_setup.getStateSpace()
+        for xy in [(-0.5, 0.4), (0.1, 0.2), (0.3, -0.6)]:
+            state = ob.State(space)
+            state().setXY(*xy)
+            states.append(state)
+        max_cost = max(
+            distance_objective.ompl_path_objective.motionCost(s1(), s2()).value()
+            for s1, s2 in itertools.combinations(iterable=states, r=2)
+        )
+        assert distance_objective.find_maximum_motion_cost([s() for s in states]) == pytest.approx(
+            max_cost / distance_objective.max_motion_cost
+        )
```

Comment:
What do you think about adding an explicit test for the other objectives? Only need to test one method per objective

Reply:
Sure but I didn't understand why we are doing this. I think we are checking if find_maximum_motion_cost() gives the same output as max() for three samples. But find_maximum_motion_cost() itself computes it using max() function right?

Reply:
Yea this seems unnecessary, removed